src/contracts/deploymentHelper/DeploymentHelper.sol

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+pragma solidity ^0.8.10;
+
+import "../vendored/interfaces/MultiTokenMediatorInterface.sol";
+import "../CowProtocolVirtualToken.sol";
+
+/// @dev Contract contains the logic for deploying CowProtocolVirtualToken on gnosis chain
+/// @author CoW Protocol Developers
+contract DeploymentHelper {
+    address public immutable foreignToken;
+    address public immutable multiTokenMediator;
+    bytes32 public immutable merkleRoot;
+    address payable public communityFundsTarget;
+    address public immutable gnoToken;
+    uint256 public immutable gnoPrice;
+    address public immutable wrappedNativeToken;
+    uint256 public immutable nativeTokenPrice;
+
+    constructor(
+        address _foreignToken,
+        address _multiTokenMediator,
+        bytes32 _merkleRoot,
+        address payable _communityFundsTarget,
+        address _gnoToken,
+        uint256 _gnoPrice,
+        address _wrappedNativeToken,
+        uint256 _nativeTokenPrice
+    ) {
+        foreignToken = _foreignToken;
+        multiTokenMediator = _multiTokenMediator;
+        merkleRoot = _merkleRoot;
+        communityFundsTarget = _communityFundsTarget;
+        gnoToken = _gnoToken;
+        gnoPrice = _gnoPrice;
+        wrappedNativeToken = _wrappedNativeToken;
+        nativeTokenPrice = _nativeTokenPrice;
+    }
+
+    function deploy() external returns (address) {
+        address bridgedToken = MultiTokenMediatorInterface(multiTokenMediator)
+            .bridgedTokenAddress(foreignToken);
+
+        // This requirement ensures that the CowProtocolVirtualToken can not be deployed
+        // with the correct cowToken, before the cowToken has been bridged by the gnosisDAO.
+        // Hence, the gnosisDAO is activating the deployment process
```

Comment:
```suggestion
        // A call to this function will be triggered by another transaction from
        // the Ethereum chain. The COW token will not be deployed in the chain
        // that hosts this contract, but it will be a bridged token from the
        // Ethereum chain. Because of this, we need the token to be available
        // before the virtual token can be deployed.
```